ui-tests/src/test/java/io/jenkins/plugins/coverage/CoverageReportTest.java

```diff
@@ -0,0 +1,309 @@
+        CoverageReportTest.verifyFileCoverageTableContent(coverageTable,
+                new String[] {"edu.hm.hafner.util", "edu.hm.hafner.util", "edu.hm.hafner.util"},
+                new String[] {"Ensure.java", "FilteredLog.java", "Generated.java"},
+                new String[] {"80.00%", "100.00%", "n/a"},
+                new String[] {"86.96%", "100.00%", "n/a"});
+
+        String coverageTree = report.getCoverageTree();
+        CoverageReportTest.verifyCoverageTreeAfterSomeBuildsWithReports(coverageTree);
+
+        String coverageOverview = report.getCoverageOverview();
+        CoverageReportTest.verifyCoverageOverviewAfterSomeBuildsWithReports(coverageOverview);
+
+        String trendChart = report.getCoverageTrend();
+        TrendChartTestUtil.verifyTrendChart(trendChart, 1, 2);
+
+    }
+
+    /**
+     * Test for CoverageReport after first build with a report. Verifies charts (TrendChart, CoverageTree and
+     * CoverageOverview) as well as {@link FileCoverageTable} are being displayed and verifies some of its data.
+     */
+    @Test
+    public void testCoverageReportAfterOneBuildWithReport() {
+
+        Build secondBuild = buildSuccessfully(getJobWithReportInConfiguration());
+
+        CoverageReport report = new CoverageReport(secondBuild);
+        report.open();
+
+        FileCoverageTable coverageTable = report.getCoverageTable();
+        CoverageReportTest.verifyFileCoverageTableNumberOfMaxEntries(coverageTable, 307);
+
+        String trendChart = report.getCoverageTrend();
+        TrendChartTestUtil.verifyTrendChartContainsOnlyOneRecord(trendChart);
+
+        String coverageTree = report.getCoverageTree();
+        CoverageReportTest.verifyCoverageTreeAfterOneBuildWithReport(coverageTree);
+
+        String coverageOverview = report.getCoverageOverview();
+        CoverageReportTest.verifyCoverageOverviewAfterOneBuildWithReport(coverageOverview);
+
+    }
+
+    /**
+     * Test for CoverageTable which should contain multiple pages. Verfies {@link FileCoverageTable} with different
+     * pages works correctly, by checking some of its {@link FileCoverageTableRow}s on different pages.
+     */
+    @Test
+    public void testCoverageTableWithMultiplePages() {
+
+        Build secondBuild = buildSuccessfully(getJobWithReportInConfiguration());
+
+        CoverageReport report = new CoverageReport(secondBuild);
+        report.open();
+
+        FileCoverageTable table = report.openFileCoverageTable();
+        CoverageReportTest.verifyFileCoverageTableContent(table,
+                new String[] {"edu.hm.hafner.analysis.parser.dry", "edu.hm.hafner.analysis", "edu.hm.hafner.analysis.parser.violations"},
+                new String[] {"AbstractDryParser.java", "AbstractPackageDetector.java", "AbstractViolationAdapter.java"},
+                new String[] {"85.71%", "88.24%", "91.67%"},
+                new String[] {"83.33%", "50.00%", "100.00%"});
```

Comment:
It would be more readable if this code would use AssertJ features:
```
assertThat(table.getRows())
      .extracting(FileCoverageTableRow::getPackage)
      .containsExaclty("edu.hm.hafner.analysis.parser.dry", 
                  "edu.hm.hafner.analysis", 
                  "edu.hm.hafner.analysis.parser.violations");
```